src/soap.ts

```diff
@@ -233,6 +234,21 @@ export class SOAP<S extends Schema> extends HttpApi<S> {
   /** @override */
   beforeSend(request: HttpRequest & { _message: object }) {
     request.body = this._createEnvelope(request._message);
+
+    const bodySize = getBodySize(request.body, request.headers);
+
+    if (
+      request.method === 'POST' &&
+      request.headers &&
+      !request.headers['transfer-encoding'] &&
+      !request.headers['content-length'] &&
+      !!bodySize
+    ) {
+      this._logger.debug(
+        `missing 'content-length' header, setting it to: ${bodySize}`,
+      );
+      request.headers['content-length'] = String(bodySize);
+    }
```

Comment:
same as this but for SOAP requests:
https://github.com/jsforce/jsforce/pull/1381/files#r1412264591

one difference here is that this only checks if method is `POST`
[while GET is possible](https://stackoverflow.com/a/26339467), I haven't seen any examples of using non-POST in API docs. We can revert it later if it's not true.